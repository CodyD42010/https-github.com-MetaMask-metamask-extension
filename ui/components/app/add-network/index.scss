.add-network {
  &__header {
    border-bottom: 1px solid var(--color-border-default);

    &__back-icon {
      margin-left: 24px;
      margin-right: 16px;
    }
  }

  &__warning-icon {
    color: var(--color-icon-muted);
  }

  &__warning-tooltip {
    color: var(--color-text-alternative);
    width: 180px;

    a {
      color: var(--color-primary-default);
    }
  }

  &__add-icon {
    color: var(--color-text-alternative);
    margin-left: auto;
    margin-right: 0;
    cursor: pointer;
  }

  &__add-button.button {
    color: var(--color-primary-default);
    font-size: $font-size-h7;
    margin-left: 24px;
  }

  &__footer {
    border-top: 1px solid var(--color-border-muted);

    & .btn-link {
      display: initial;
      padding: 0;
    }

    &__link {
      color: var(--color-primary-default);
    }

    & .actionable-message--warning .actionable-message__message,
    .actionable-message--warning .actionable-message__action {
      color: var(--color-text-alternative);
    }
  }
}


