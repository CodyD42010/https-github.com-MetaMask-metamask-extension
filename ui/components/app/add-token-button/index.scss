.add-token-button {
  &__button {
    color: black !important;
    border-color: none !important;
    background-color: $secondary-dark;
    max-width: 200px;
    margin: 16px auto;
  }
}
