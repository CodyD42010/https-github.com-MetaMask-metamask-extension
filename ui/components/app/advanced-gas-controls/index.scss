.advanced-gas-controls {
  &__row {
    margin-bottom: 20px;
  }

  .info-tooltip {
    display: inline-block;
  }

  .form-field__row--error .form-field__heading-title h6 {
    color: var(--error-1);

    & path {
      fill: var(--error-1);
    }
  }

  h6 {
    padding-bottom: 6px;
    margin-inline-end: 6px;
  }

  path {
    fill: var(--ui-3);
  }
}
