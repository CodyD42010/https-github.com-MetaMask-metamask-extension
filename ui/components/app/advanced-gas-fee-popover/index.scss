.advanced-gas-fee-popover {
  .form-field__heading-title > h6 {
    font-size: $font-size-h7;
  }

  .popover-header {
    border-radius: 0;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
    border-bottom: 1px solid var(--ui-grey);
  }

  .popover-footer {
    border-top: none;
  }
}
