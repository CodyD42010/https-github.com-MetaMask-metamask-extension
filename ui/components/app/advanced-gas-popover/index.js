export { default } from './advanced-gas-popover';
