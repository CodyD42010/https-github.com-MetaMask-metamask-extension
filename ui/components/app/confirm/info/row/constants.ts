export const TEST_ADDRESS = '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1';
