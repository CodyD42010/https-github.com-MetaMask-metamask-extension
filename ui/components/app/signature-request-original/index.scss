.request-signature {
  &__container {
    width: 380px;
    border-radius: 8px;
    background-color: var(--color-background-default);
    box-shadow: var(--shadow-size-xs) var(--color-shadow-default);
    display: flex;
    flex-flow: column nowrap;
    z-index: 25;
    align-items: center;
    position: relative;
    height: 100%;

    @include screen-sm-max {
      width: 100%;
      top: 0;
      box-shadow: none;
    }

    @include screen-sm-min {
      height: 620px;
    }

    &__reject {
      padding-bottom: 20px;
    }
  }

  &__typed-container {
    padding: 17px;

    h1 {
      font-weight: 900;
      margin-bottom: 5px;
    }

    * {
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    > div {
      margin-bottom: 10px;
    }
  }

  &__content__title {
    @include H3;

    display: flex;
    justify-content: center;
    color: var(--color-default-text);
    z-index: 3;
  }

  &__account {
    width: 100%;
  }

  &__body {
    width: 100%;
    height: 100%;
    display: flex;
    flex-flow: column;
    flex: 1 1 auto;
  }

  &__origin {
    display: flex;
    justify-content: center;
    margin-top: 16px;
    margin-bottom: 16px;
  }

  &__notice,
  &__warning {
    @include H6;

    text-align: center;
    margin-top: 41px;
    margin-bottom: 11px;
    width: 100%;
  }

  &__notice {
    color: var(--color-text-alternative);
    padding: 0 10px;
  }

  &__rows {
    height: 100%;
    overflow-y: auto;
    overflow-x: hidden;
    border-top: 1px solid var(--color-border-default);
    display: flex;
    flex-flow: column;
  }

  &__row {
    display: flex;
    flex-flow: column;
    flex: 1 0 auto;
  }

  &__row-title {
    @include H5;

    width: 80px;
    color: var(--color-text-alternative);
    margin-top: 12px;
    margin-left: 18px;
    width: 100%;
  }

  &__row-value {
    @include H6;

    color: var(--color-text-alternative);
    width: 100%;
    overflow-wrap: break-word;
    border-bottom: 1px solid var(--color-border-default);
    padding: 6px 18px 15px;
    white-space: pre-wrap;
  }

  &__help-link {
    cursor: pointer;
    text-decoration: underline;
    color: var(--color-primary-default);
    margin-inline-start: 3px;
  }

  &__footer {
    @include H3;

    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    position: relative;
    flex: 0 0 auto;
    border-top: 1px solid var(--color-border-default);
    padding: 1.6rem;

    button {
      width: 165px;
    }

    &__cancel-button {
      margin-right: 1.2rem;
    }
  }
}
