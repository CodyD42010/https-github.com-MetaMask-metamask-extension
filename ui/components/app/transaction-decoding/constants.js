const TX_INSIGHTS_BASE_URI =
  'https://tx-insights-api.metaswap-dev.codefi.network';

export const FETCH_PROJECT_INFO_URI = `${TX_INSIGHTS_BASE_URI}/fetch-project`;
export const FETCH_SUPPORTED_NETWORKS_URI = `${TX_INSIGHTS_BASE_URI}/networks`;
