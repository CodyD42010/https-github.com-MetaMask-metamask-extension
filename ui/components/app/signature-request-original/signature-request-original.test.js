import React from 'react';
import configureMockStore from 'redux-mock-store';
import { fireEvent, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { MESSAGE_TYPE } from '../../../../shared/constants/app';
import { SECURITY_PROVIDER_MESSAGE_SEVERITY } from '../../../../shared/constants/security-provider';
import mockState from '../../../../test/data/mock-state.json';
import { renderWithProvider } from '../../../../test/lib/render-helpers';
import configureStore from '../../../store/store';
import {
  resolvePendingApproval,
  rejectPendingApproval,
  completedTx,
} from '../../../store/actions';
import SignatureRequestOriginal from '.';

jest.mock('../../../store/actions', () => ({
  resolvePendingApproval: jest.fn().mockReturnValue({ type: 'test' }),
  rejectPendingApproval: jest.fn().mockReturnValue({ type: 'test' }),
  completedTx: jest.fn().mockReturnValue({ type: 'test' }),
}));

const MOCK_SIGN_DATA = JSON.stringify({
  domain: {
    name: 'happydapp.website',
  },
  message: {
    string: 'haay wuurl',
    number: 42,
  },
  primaryType: 'Mail',
  types: {
    EIP712Domain: [
      { name: 'name', type: 'string' },
      { name: 'version', type: 'string' },
      { name: 'chainId', type: 'uint256' },
      { name: 'verifyingContract', type: 'address' },
    ],
    Group: [
      { name: 'name', type: 'string' },
      { name: 'members', type: 'Person[]' },
    ],
    Mail: [
      { name: 'from', type: 'Person' },
      { name: 'to', type: 'Person[]' },
      { name: 'contents', type: 'string' },
    ],
    Person: [
      { name: 'name', type: 'string' },
      { name: 'wallets', type: 'address[]' },
    ],
  },
});

const props = {
  signMessage: jest.fn(),
  cancelMessage: jest.fn(),
  txData: {
    msgParams: {
      from: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
      data: MOCK_SIGN_DATA,
      origin: 'https://happydapp.website/governance?futarchy=true',
    },
    type: MESSAGE_TYPE.ETH_SIGN,
  },
  selectedAccount: {
    address: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
  },
};

const render = (txData = props.txData) => {
  const store = configureStore({
    metamask: {
      ...mockState.metamask,
    },
  });

  return renderWithProvider(
    <SignatureRequestOriginal {...props} txData={txData} />,
    store,
  );
};

describe('SignatureRequestOriginal', () => {
  const store = configureMockStore()(mockState);

  it('should match snapshot', () => {
    const { container } = renderWithProvider(
      <SignatureRequestOriginal {...props} />,
      store,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render navigation', () => {
    render();
    const navigationContainer = screen.queryByTestId('navigation-container');
    expect(navigationContainer).toBeInTheDocument();
  });

  it('should render eth sign screen', () => {
    render();
    expect(screen.getByText('Signature request')).toBeInTheDocument();
  });

  it('should render warning for eth sign when sign button clicked', async () => {
    render();
    const signButton = screen.getByTestId('page-container-footer-next');

    fireEvent.click(signButton);
    expect(screen.getByText('Your funds may be at risk')).toBeInTheDocument();

    const secondSignButton = screen.getByTestId(
      'signature-warning-sign-button',
    );
    await act(async () => {
      fireEvent.click(secondSignButton);
    });
    expect(resolvePendingApproval).toHaveBeenCalledTimes(1);
    expect(completedTx).toHaveBeenCalledTimes(1);
  });

  it('should cancel approval when user reject signing', async () => {
    render();
    const rejectButton = screen.getByTestId('page-container-footer-cancel');
    await act(async () => {
      fireEvent.click(rejectButton);
    });
    expect(rejectPendingApproval).toHaveBeenCalledTimes(1);
  });

  it('should escape RTL character in label or value', () => {
    const txData = {
      msgParams: {
        from: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
        data: [
          {
            type: 'string',
            name: 'Message \u202E test',
            value: 'Hi, \u202E Alice!',
          },
        ],
        origin: 'https://happydapp.website/governance?futarchy=true',
      },
      type: MESSAGE_TYPE.ETH_SIGN_TYPED_DATA,
    };
    const { getByText } = render(txData);
    expect(getByText('Message \\u202E test:')).toBeInTheDocument();
    expect(getByText('Hi, \\u202E Alice!')).toBeInTheDocument();
  });

  it('should render SecurityProviderBannerMessage component properly', () => {
    props.txData.securityProviderResponse = {
      flagAsDangerous: '?',
      reason: 'Some reason...',
      reason_header: 'Some reason header...',
    };
    render();
    expect(screen.getByText('Request not verified')).toBeInTheDocument();
    expect(
      screen.getByText(
        'Because of an error, this request was not verified by the security provider. Proceed with caution.',
      ),
    ).toBeInTheDocument();
    expect(screen.getByText('OpenSea')).toBeInTheDocument();
  });

  it('should not render SecurityProviderBannerMessage component when flagAsDangerous is not malicious', () => {
    props.txData.securityProviderResponse = {
      flagAsDangerous: SECURITY_PROVIDER_MESSAGE_SEVERITY.NOT_MALICIOUS,
    };

    render();
    expect(screen.queryByText('Request not verified')).toBeNull();
    expect(
      screen.queryByText(
        'Because of an error, this request was not verified by the security provider. Proceed with caution.',
      ),
    ).toBeNull();
    expect(screen.queryByText('OpenSea')).toBeNull();
  });
});
