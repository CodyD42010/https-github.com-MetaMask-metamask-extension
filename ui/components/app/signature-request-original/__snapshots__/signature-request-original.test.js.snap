// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignatureRequestOriginal should match snapshot 1`] = `
<div>
  <div
    class="request-signature__container"
  >
    <div
      class="request-signature__account"
    >
      <div
        class="box network-account-balance-header box--padding-4 box--display-flex box--flex-direction-row box--justify-content-space-between box--align-items-center"
      >
        <div
          class="box box--display-flex box--gap-2 box--flex-direction-row box--align-items-center"
        >
          <div
            class="box box--display-flex box--flex-direction-row box--align-items-center"
          >
            <div
              class=""
            >
              <div
                class="identicon"
                style="height: 32px; width: 32px; border-radius: 16px;"
              >
                <div
                  style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 32px; height: 32px; display: inline-block; background: rgb(250, 58, 0);"
                >
                  <svg
                    height="32"
                    width="32"
                    x="0"
                    y="0"
                  >
                    <rect
                      fill="#18CDF2"
                      height="32"
                      transform="translate(-1.04839350379394 -3.3042840694604987) rotate(328.9 16 16)"
                      width="32"
                      x="0"
                      y="0"
                    />
                    <rect
                      fill="#035E56"
                      height="32"
                      transform="translate(-18.298461708832043 10.5924618717486) rotate(176.2 16 16)"
                      width="32"
                      x="0"
                      y="0"
                    />
                    <rect
                      fill="#F26602"
                      height="32"
                      transform="translate(16.667842018223922 -14.205139722997082) rotate(468.9 16 16)"
                      width="32"
                      x="0"
                      y="0"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="network-account-balance-header__network-account__ident-icon-ethereum--gray"
            >
              <span
                class="icon-with-fallback__fallback"
              >
                U
              </span>
            </div>
          </div>
          <div
            class="box box--display-flex box--flex-direction-column box--align-items-flex-start"
          >
            <h6
              class="box box--margin-top-1 box--flex-direction-row typography typography--h6 typography--weight-normal typography--style-normal typography--color-text-alternative"
            >
              Unknown private network
            </h6>
            <h6
              class="box box--margin-bottom-1 box--flex-direction-row typography typography--h6 typography--weight-bold typography--style-normal typography--color-text-default"
            >
              Test Account
            </h6>
          </div>
        </div>
        <div
          class="box box--display-flex box--flex-direction-column box--align-items-flex-end"
        >
          <h6
            class="box box--margin-top-1 box--flex-direction-row typography typography--h6 typography--weight-normal typography--style-normal typography--color-text-alternative"
          >
            Balance
          </h6>
          <h6
            class="box box--margin-bottom-1 box--flex-direction-row typography typography--h6 typography--weight-bold typography--style-normal typography--align-end typography--color-text-default"
          >
            0
             
            ETH
          </h6>
        </div>
      </div>
    </div>
    <div
      class="request-signature__body"
    >
      <div
        class="request-signature__origin"
      >
        <div
          class="site-origin"
        >
          <div
            class="chip chip--with-left-icon chip--border-color-border-muted chip--background-color-undefined"
          >
            <div
              class="chip__left-icon"
            >
              <div
                class=""
              >
                <span
                  class="icon-with-fallback__fallback"
                />
              </div>
            </div>
            <span
              class="box box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography chip__label typography--h6 typography--weight-normal typography--style-normal typography--color-text-alternative"
            >
              https://happydapp.website/governance?futarchy=true
            </span>
          </div>
        </div>
      </div>
      <h3
        class="box box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography request-signature__content__title typography--h3 typography--weight-bold typography--style-normal typography--color-text-default"
      >
        Signature request
      </h3>
      <h6
        class="box box--margin-12 box--margin-top-3 box--margin-bottom-1 box--flex-direction-row typography request-signature__content__subtitle typography--h7 typography--weight-normal typography--style-normal typography--align-center typography--color-text-alternative"
      >
        Only sign this message if you fully understand the content and trust the requesting site.
      </h6>
      <div
        class="request-signature__notice"
      >
        You are signing:
      </div>
      <div
        class="request-signature__rows"
      >
        <div
          class="request-signature__row"
        >
          <div
            class="request-signature__row-title"
          >
            Message:
          </div>
          <div
            class="request-signature__row-value"
          >
            {"domain":{"name":"happydapp.website"},"message":{"string":"haay wuurl","number":42},"primaryType":"Mail","types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Group":[{"name":"name","type":"string"},{"name":"members","type":"Person[]"}],"Mail":[{"name":"from","type":"Person"},{"name":"to","type":"Person[]"},{"name":"contents","type":"string"}],"Person":[{"name":"name","type":"string"},{"name":"wallets","type":"address[]"}]}}
          </div>
        </div>
      </div>
    </div>
    <div
      class="request-signature__footer"
    >
      <button
        class="button btn--rounded btn-secondary btn--large request-signature__footer__cancel-button"
        role="button"
        tabindex="0"
      >
        Reject
      </button>
      <button
        class="button btn--rounded btn-primary btn--large request-signature__footer__sign-button"
        data-testid="request-signature__sign"
        role="button"
        tabindex="0"
      >
        Sign
      </button>
    </div>
  </div>
</div>
`;
